src/main/java/org/openrewrite/staticanalysis/DefaultComesLastVisitor.java

```diff
@@ -41,117 +45,140 @@ public J.Switch visitSwitch(J.Switch switch_, P p) {
+                if (defaultCaseFound) {
+                    defaultCases.addAll(fallThroughCases);
+                    fallThroughCases.clear();
+                    defaultCaseFound = false;
                 } else {
-                    if (defaultCase != null && defaultCase.getStatements().isEmpty() && !foundNonEmptyCase) {
-                        // the last case grouped with default can be non-empty. it will be flipped with
-                        // the default case, including its statements
-                        casesGroupedWithDefault.add(aCase);
-                    }
-                    foundNonEmptyCase = true;
-                    fixedCases.add(aCase);
+                    casesWithDefaultLast.addAll(fallThroughCases);
+                    fallThroughCases.clear();
                 }
             }
+        }
 
-            if (defaultCase != null && !casesGroupedWithDefault.isEmpty()) {
-                J.Case lastGroupedWithDefault = casesGroupedWithDefault.get(casesGroupedWithDefault.size() - 1);
-                if (!lastGroupedWithDefault.getStatements().isEmpty()) {
-                    casesGroupedWithDefault.set(casesGroupedWithDefault.size() - 1,
-                            lastGroupedWithDefault.withStatements(Collections.emptyList()));
-                    defaultCase = defaultCase.withStatements(lastGroupedWithDefault.getStatements());
-                }
-            }
+        casesWithDefaultLast = addBreakToLastCase(casesWithDefaultLast, p);
+        casesWithDefaultLast.addAll(maybeReorderFallthroughCases(defaultCases, p));
+        casesWithDefaultLast = ListUtils.mapLast(casesWithDefaultLast, this::removeBreak);
+        return casesWithDefaultLast;
+    }
 
-            J.Case lastNotGroupedWithDefault = fixedCases.get(fixedCases.size() - 1);
-            if (!lastNotGroupedWithDefault.getStatements().stream().reduce((s1, s2) -> s2)
-                    .map(stat -> stat instanceof J.Break || stat instanceof J.Continue ||
-                                 stat instanceof J.Return || stat instanceof J.Throw)
-                    .orElse(false)) {
-
-                // add a break statement since this case is now no longer last and would fall through
-                List<Statement> statementsOfCaseBeingMoved = new ArrayList<>(lastNotGroupedWithDefault.getStatements());
-                J.Break breakStatement = autoFormat(
-                        new J.Break(Tree.randomId(), Space.EMPTY, Markers.EMPTY, null),
-                        p
-                );
-                statementsOfCaseBeingMoved.add(breakStatement);
-
-                lastNotGroupedWithDefault = lastNotGroupedWithDefault.withStatements(
-                        ListUtils.map(statementsOfCaseBeingMoved, stmt -> autoFormat(stmt, p, getCursor()))
-                );
-                fixedCases.set(fixedCases.size() - 1, lastNotGroupedWithDefault);
+    private @NonNull List<J.Case> maybeReorderFallthroughCases(List<J.Case> cases, P p) {
```

Comment:
```suggestion
    private List<J.Case> maybeReorderFallthroughCases(List<J.Case> cases, P p) {
```

No need for this annotation, as there is already a JSpecify package-info.java file in this folder.